streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -156,12 +174,21 @@ public long headRecordTimestamp() {
         return headRecord == null ? UNKNOWN : headRecord.timestamp;
     }
 
+    public long headRecordOffsetPlusOneOrConsumerPosition() {
+        if (consumerPosition < 0) {
+            throw new IllegalStateException("You must call `addRecords(0` at least once (after `RecordQueue` " +
+                "creation or after calling `clear()`, before you can call `headRecordOffsetOrConsumerPosition()`.");
+        }
+        return headRecord != null ? headRecord.offset() + 1 : consumerPosition;
```

Comment:
Instead of returning `headRecord.offset()` we need to add `+ 1` to it (the called cannot know if "offset" or "position" was returned (I changed the return type from `Long` to `long` and we don't return `null` any longer).